index.html

```diff
@@ -762,18 +838,126 @@ <h2><dfn>CaptureController</dfn></h2>
+                      is set even if {{MediaDevices/getDisplayMedia()}} rejects.
+                    </div>
+                  </p>
+                </li>
+                <li>
+                  <p>
+                    Create one internal slot: <dfn data-dfn-for="CaptureController">[[\initialFocusLost]]</dfn>,
+                    initialized to <code>false</code>.
+                    <div class="note">
+                      This internal slot ensures that calls to {{CaptureController/setFocusBehavior()}} fail
+                      without throwing if the user manually focuses another tab or native application.
+                    </div>
+                  </p>
+                </li>
+                <li>
+                  <p>
+                    Create one internal slot: <dfn data-dfn-for="CaptureController">[[\focusWindowClosed]]</dfn>,
+                    initialized to <code>false</code>.
+                    <div class="note">
+                      This internal slot tracks explicit closure of the window of opportunity, which happens when
+                      {{CaptureController/setFocusBehavior()}} is called, or the task to close the window executes.
+                      As applications can easily avoid such errors, {{CaptureController/setFocusBehavior()}} when
+                      this slot is set to <code>true</code> throw.
+                    </div>
+                  </p>
+                </li>
+              </ol>
+            </dd>
+            <dt>
+              <dfn>setFocusBehavior</dfn>
+            </dt>
             <dd>
               <p>
-                Create one internal slot: <dfn data-dfn-for="CaptureController">[[\isBound]]</dfn>,
-                initialized to <code>false</code>.
+                Run the following steps:
               </p>
+              <ol>
+                <li>
+                  <p>
+                    If {{CaptureController/[[isBound]]}} is <code>false</code>,
+                    [=exception/throw=] an "{{InvalidStateError}}" {{DOMException}}.
+                  </p>
+                </li>
+                <li>
+                  <p>
+                    If {{CaptureController/[[videoTrack]]}} is <code>null</code>,
+                    [=exception/throw=] a "{{InvalidStateError}}" {{DOMException}}.
+                  </p>
+                </li>
+                <li>
+                  <p>
+                    If {{CaptureController/[[videoTrack]]}}.{{MediaStreamTrack/getSettings()}}.{{displaySurface}}
+                    is neither {{DisplayCaptureSurfaceType/"browser"}} nor {{DisplayCaptureSurfaceType/"window"}},
+                    [=exception/throw=] a "{{NotAllowedError}}" {{DOMException}}.
+                  </p>
+                </li>
+                <li>
+                  <p>
+                    <!-- TODO: Fix the ReadyState reference. -->
+                    If {{CaptureController/[[videoTrack]]}}'s
```

Comment:
I am interested in calling the method ahead of time. I'm also wondering why the method isn't idempotent?

I'm curious, why did we reject a setter instead of a method? E.g.

```js
const captureController = new CaptureController;
captureController.focusBehavior = "no-focus-change";
navigator.mediaDevices({video: true, controller: captureController});
```
To clarify, I'm a fan of using the right JS surface where applicable. I was a fan of method when it was a simple `focus()` method, and I consider "set-methods" a bit of an anti-pattern really. But if both of you have good reasons for that that you can share, I'm willing to compromise.

Reply:
> I am interested in calling the method ahead of time.

Okay, then let's add that as a follow-up PR, which I'll produce and upload for review after the current PR gets merged.

> I'm also wondering why the method isn't idempotent?

Because I don't expect this to be called more than once by a well-structured app, given how it can only be called within a very narrow window of opportunity. But the follow-up will change things so that we can call arbitrarily many times before gDM is called, with same/different values, and up to one time after gDM is called.

> I'm curious, why did we reject a setter instead of a method?

It's more natural for a method to (1) have side-effect and (2) throw when used at an inappropriate time.

Reply:
Now that Jan-Ivar has migrated this discussion and I have copied over my responses, I am resolving this thread.